### PR TITLE
Fix weather particle physics.

### LINKED ISSUE #18025: Weather effect faster with higher framerate
Snow particles move much faster with higher framerate.
Tested with Playtest 20200426 and Releasse 20200503, on 60Hz Monitor, Win10 and over 900 FPS. Also notable with 120 FPS. I guess it's the same with higher refresh rate Panels.

Reproduce:

- Disable VSync and Frame Limiter in Settings
- Start Game on Snow Themed Map with Snow a weather effect
- With 900 Frames the Snow particle jump around

### PR DESCRIPTION
**Pull request by pchote**
Fixes #18025. See commit message for description.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs [hunk 1]
```diff
@@ -197,6 +198,12 @@ void IRenderAboveWorld.RenderAboveWorld(Actor self, WorldRenderer wr)
 			var viewport = new Rectangle(center - new int2(viewportSize) / 2, viewportSize);
 			var wcr = Game.Renderer.WorldRgbaColorRenderer;
 
+			// SwingSpeed is defined in px/tick so we must account for the fraction of a tick that elapsed since the last render.
+			// The scale is capped at 1 tick to avoid unexpected behaviour at game start, if RunTime overflows, or if the game stalls.
+			var runtime = Game.RunTime;
+			var tickFraction = Math.Min((runtime - lastRender) * 1f / world.Timestep, 1);
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs:204]: `tickFraction` looks like something we might want to eventually calculate and expose at a central, publicly accessible place like `Game`, so we can reuse it for things like interpolation and don't need to do the math locally in every place that needs it.

Just thinking aloud, not a request for this particular PR.

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs:204]: You are thinking of something like https://love2d.org/wiki/dt delta time?

**Review by Smittytron — APPROVED** [review]
Fix confirmed and LGTM

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/60d159286bc5da014ea9be63497f78ceab283d92...3911fde2ae8da1b53d062cd9a8e2520bb32709e6

**Comment by TheRaffy** [comment]
Hey everybody, thanks for the fix!

**Pull request merged**